### PR TITLE
Autodetect terminal capabilities, update flags

### LINKED ISSUE #70: Feature request: Detect terminal feature support
xcpretty could use something like `tput colors` to decide whether to its output should use colors or not. In a similar vein, `locale charmap` could be used to choose a default output encoding.

Just a suggestion. Great project!

### PR DESCRIPTION
**Pull request by kattrali**
Adds support for terminal capability detection for unicode and color, enabling where detected.

Fixes #70 

Thoughts?

### CODE REVIEW

File: bin/xcpretty [hunk 1]
```diff
@@ -19,11 +19,22 @@ report_formats = {
...
 
 printer_opts = {
-  :unicode => true,
-  :colorize => false,
+  :unicode => XCPretty::Term.unicode?,
+  :colorize => XCPretty::Term.color?,
   :formatter => XCPretty::Simple
 }
 
+def opt_bool(field, value)
```
Comment (supermarin) [inline bin/xcpretty]: `bool_option`?

Comment (kattrali) [inline bin/xcpretty]: That makes far more sense. I've been writing too much C.

**Comment by kattrali** [comment]
Updated.

**Comment by supermarin** [comment]
:+1: :+1: after comment

**Comment by kattrali** [comment]
Are we good here?

**Comment by supermarin** [comment]
Yep :) sorry I've gotten used to giving thumbs and expecting the author to merge

**Comment by angerman** [comment]
Shouldn't these changes be reflected in the readme as well? I still find `-c` without any explanation what `-c` does.

**Comment by supermarin** [comment]
@angerman they should once we announce it in a release.

Would you like to submit a PR?

**Pull request merged**